### PR TITLE
[Reverted] Manual tutorial completion

### PR DESCRIPTION
**Pull request by yutamago**
**TL;DR**
You can now control what happens, when the tutorial hasn't been completed before.

**Changes made:**
- Added class **Tutorial** with Builder
- Added **PokemonGo#setTutorial(Tutorial)**
- Added example **TutorialCompletionExample**
- Changed methods in PlayerProfile to accept values instead of generating random values
- Changed PokemonGo#initialize() to use PokemonGo#Tutorial

**How to use**
1. Get a TutorialBuilder: `TutorialBuilder builder = Tutorial.newBuilder()`
2. Set some values (or don't) `builder.setNickname(() -> "SomeName")`
3. Values you don't set, will be generated randomly
4. Build it! `Tutorial tutorial = builder.build()`
5. Pass the tutorial to your API object `go.setTutorial(tutorial)`

**What can you set**
- whether or not you want to accept the TOS of Niantic
- how your avatar should look like 
- which nickname you want
- which starter Pokémon you want

You can also ask your users what they want. Take a look at the TutorialCompletionExample or ask me your questions if you're not sure how.

### CODE REVIEW

**Comment by yutamago** [comment]
I just noticed this was actually planned in #667 #660.

**Comment by stirante** [comment]
Isn't it a bit better to make it more like builder rather than class you need to extends?

**Comment by yutamago** [comment]
I wasn't that familiar with builders, but I can rewrite it.

**Comment by stirante** [comment]
Well, basiclly, after constructor you add a lot of methods to set something which return the builder itself. If you use Intellij Idea, it has auto generating this based on constructor parameters.

**Comment by dojo1017** [comment]
Do not merge yet. I'm currently rewriting it.

**Comment by dojo1017** [comment]
@stirante 
I put everything inside a **TutorialBuilder**, that can build a **Tutorial**.

However, you now have to provide a **TutorialProvider** to PokemonGo, that implements TutorialProvider.getTutorial():Tutorial

Without the TutorialProvider you would have to set values before login. In other words, you would have to ask your users about which nickname they want, before you know if it's even necessary or not.

I will now update the PR description.

**Comment by dojo1017** [comment]
DO NOT MERGE. I'm simplifying this, so it's easier to use.

**Comment by yutamago** [comment]
I reworked the whole thing. Now it's really easy to use, I suppose.
